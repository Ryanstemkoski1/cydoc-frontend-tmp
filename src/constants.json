{
  "medicalhistory": {
    "conditions": ["Type II Diabetes", "Myocardial Infarction", "Hypertension",  "Hypercholesteremia", "Depression", "HIV"]
  },

  "socialhistory": {
    "substanceUseContentHeader": "Substance Use",
    "secondaryFields": ["living situation", "employment", "diet", "exercise"],
    "substanceAbuseFields" : {
      "tobacco": {
        "condition": "tobacco",
        "firstField": "Packs/Day",
        "secondField": "Number of Years"
      },
      "alcohol": {
        "condition": "alcohol",
        "firstField": "Drinks/Week",
        "secondField": "What kind of drinks?"
      },
      "substanceAbuse": {
        "condition": "substance abuse",
        "firstField": "Drug Name",
        "secondField": "Mode of Delivery"
      }
    }
  }

}