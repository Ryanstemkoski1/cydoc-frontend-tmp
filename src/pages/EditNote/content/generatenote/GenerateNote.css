.ui.icon.button.note-previous-button {
    position: sticky !important;
    bottom: 0;
    opacity: 0.8;
    margin: 0;
    display: block;
    font-size: 13px;
    background: #c8e2ea;
  }
  
  .ui.icon.button.note-previous-button:hover {
    opacity: 1;
    background: #c8e2ea;
  }
  
  .ui.icon.button.note-previous-button {
    margin-left: -13em;
  }
  
  .ui.icon.button.small-note-previous-button {
    display: none;
    position: sticky !important;
    bottom: 0;
    opacity: 0.8;
    margin: 0;
    background: #c8e2ea;
  }
  
  @media screen and (max-width: 1300px) {
    .ui.icon.button.note-previous-button {
        display: none;
    }
  
    .ui.icon.button.small-note-previous-button {
        display: block;
    }
    
    .ui.icon.button.small-note-previous-button {
      margin-left: -2.8em;
    }
  
  }
  
  @media screen and (max-width: 400px) {
    .ui.icon.button.small-note-previous-button {
      margin-left: 0;
    }
  
    .ui.icon.button.small-note-next-button {
      margin-right: 0;
    }
  }
  
