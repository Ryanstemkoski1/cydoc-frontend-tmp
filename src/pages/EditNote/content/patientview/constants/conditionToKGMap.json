{
    "ADHD": "ADHD",
    "Depression": "DEPRESSION",
    "Genital Symptoms": "SEXUALINFECTION",
    "High Blood Pressure": "HYPERTENSION",
    "High Cholesterol": "HYPERLIPIDEMIA",
    "High Triglycerides": "HYPERLIPIDEMIA",
    "Urinary Symptoms": "URINARY_SYMPTOMS"
}
