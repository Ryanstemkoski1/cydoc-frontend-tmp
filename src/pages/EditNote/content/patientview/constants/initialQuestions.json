{
    "order": {
        "1": "1",
        "2": "2",
        "3": "3",
        "4": "4",
        "5": "5",
        "6": "6",
        "7": "7"
    },
    "graph": {
        "1": ["2", "3", "4", "8", "9"],
        "2": [],
        "3": [],
        "4": [],
        "5": ["6", "7"],
        "6": [],
        "7": []
    },
    "nodes" : {
            "1": {
                "uid": "1",
                "text": "",
                "responseType": "",
                "category": "",
                "doctorView": ""
            },
            "2": {
                "uid": "2",
                "text": "Are you here for an annual physical exam?",
                "responseType": "YES-NO",
                "category": "ANNUALVISIT",
                "doctorView": "Annual Physical Exam"
            },
            "3": {
                "uid": "3",
                "text": "Are you here to follow up any previously diagnosed medical conditions?",
                "responseType": "YES-NO",
                "category": "",
                "doctorView": ""
            },
            "4": {
                "uid": "4",
                "text": "Do you have any new health concerns to discuss with the clinician today?",
                "responseType": "YES-NO",
                "category": "",
                "doctorView": ""
            },
            "5": {
                "uid": "5",
                "text": "",
                "responseType": "",
                "category": "",
                "doctorView": ""
            },
            "6": {
                "uid": "6",
                "text": "Please select the most important conditions or symptoms that you would like to discuss today, up to a maximum of 3. CLICK[ADHD, Depression, Diabetes, Genital Symptoms/STI, Hypertension, Hyperlipidemia, Urinary Symptoms/UTI]",
                "responseType": "SELECTONE",
                "category": "",
                "doctorView": ""
            },
            "7": {
                "uid": "7",
                "text": "If youâ€™d like to discuss an additional condition or symptom not listed above, please search for it here:",
                "responseType": "SEARCH",
                "category": "",
                "doctorView": ""
            },
            "8": {
                "uid": "8",
                "text": "Please confirm the date of your appointment.",
                "responseType": "TIME3DAYS",
                "category": "",
                "doctorView": ""
            },
            "9": {
                "uid": "9",
                "text": "What is the last name of the clinician you are seeing today? (optional)",
                "responseType": "SHORT-TEXT",
                "category": "",
                "doctorView": ""
            }
        }
}
