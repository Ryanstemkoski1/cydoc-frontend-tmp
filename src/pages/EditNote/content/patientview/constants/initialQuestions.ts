import { ResponseTypes } from 'constants/hpiEnums';
import { InitialQuestionsState } from 'redux/reducers/userViewReducer';

const initialQuestions: InitialQuestionsState = {
    order: {
        '1': '1',
        '2': '2',
        '3': '3',
        '4': '4',
        '5': '5',
        '6': '6',
        '7': '7',
    },
    graph: {
        '1': ['8', '9'],
        '2': [],
        '3': [],
        '4': [],
        '5': ['6', '7'],
        '6': [],
        '7': [],
    },
    nodes: {
        '1': {
            uid: '1',
            text: '',
            responseType: ResponseTypes.NULL,
            category: '',
            doctorView: '',
        },
        '2': {
            uid: '2',
            text: 'Are you here for an annual physical exam?',
            responseType: ResponseTypes.YES_NO,
            category: 'ANNUALVISIT',
            doctorView: 'Annual Physical Exam',
        },
        '3': {
            uid: '3',
            text: 'Are you here to follow up any previously diagnosed medical conditions?',
            responseType: ResponseTypes.YES_NO,
            category: '',
            doctorView: '',
        },
        '4': {
            uid: '4',
            text: 'Do you have any new health concerns to discuss with the clinician today?',
            responseType: ResponseTypes.YES_NO,
            category: '',
            doctorView: '',
        },
        '5': {
            uid: '5',
            text: '',
            responseType: ResponseTypes.NULL,
            category: '',
            doctorView: '',
        },
        '6': {
            uid: '6',
            text: 'Please select the most important conditions or symptoms that you would like to discuss today, up to a maximum of 3.',
            responseType: ResponseTypes.SELECTONE,
            category: '',
            doctorView: '',
        },
        '7': {
            uid: '7',
            text: 'List any other conditions or symptoms you would like to discuss with the clinician today. Please be accurate as this information will be used to organize your visit.',
            responseType: ResponseTypes.LIST_TEXT,
            category: '',
            doctorView: '',
        },
        '8': {
            uid: '8',
            text: 'Please confirm the date of your appointment.',
            responseType: ResponseTypes.TIME3DAYS,
            category: '',
            doctorView: '',
        },
        '9': {
            uid: '9',
            text: 'What is the last name of the clinician you are seeing today? (optional)',
            responseType: ResponseTypes.SHORT_TEXT,
            category: '',
            doctorView: '',
        },
    },
};

export default initialQuestions;
