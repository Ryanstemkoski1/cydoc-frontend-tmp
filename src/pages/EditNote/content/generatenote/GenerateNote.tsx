import React, { Fragment, useState, useEffect } from 'react';
import { connect } from 'react-redux';

import { CurrentNoteState } from 'redux/reducers';
import { ReviewOfSystemsState } from 'redux/reducers/reviewOfSystemsReducer';
import { selectReviewOfSystemsState } from 'redux/selectors/reviewOfSystemsSelectors';
import { selectPlanState } from 'redux/selectors/planSelectors';
import { SocialHistoryState } from 'redux/reducers/socialHistoryReducer';
import { selectFamilyHistoryState } from 'redux/selectors/familyHistorySelectors';
import { selectAllergiesState } from 'redux/selectors/allergiesSelectors';
import { AllergiesState } from 'redux/reducers/allergiesReducer';
import { selectMedicationsState } from 'redux/selectors/medicationsSelectors';
import { MedicationsState } from 'redux/reducers/medicationsReducer';
import { selectSurgicalHistoryState } from 'redux/selectors/surgicalHistorySelectors';
import { SurgicalHistoryState } from 'redux/reducers/surgicalHistoryReducer';
import { MedicalHistoryState } from 'redux/reducers/medicalHistoryReducer';
import { PhysicalExamState } from 'redux/reducers/physicalExamReducer';
import {
    Button,
    Segment,
    Input,
    Modal,
    Header,
    Form,
    Icon,
} from 'semantic-ui-react';
import { PlanState } from 'redux/reducers/planReducer';
import { FamilyHistoryState } from 'redux/reducers/familyHistoryReducer';
import { selectPatientInformationState } from 'redux/selectors/patientInformationSelector';
import { PatientInformationState } from 'redux/reducers/patientInformationReducer';
import {
    updatePatientName,
    UpdatePatientNameAction,
    updatePatientPronouns,
    UpdatePatientPronounsAction,
    UpdatePatientInformationAction,
    updatePatientInformation,
} from 'redux/actions/patientInformationActions';

// import all the individual note sections
import HPINote from './notesections/HPINote';
import MedicalHistoryNote from './notesections/MedicalHistoryNote';
import SurgicalHistoryNote from './notesections/SurgicalHistoryNote';
import MedicationsNote from './notesections/MedicationsNote';
import AllergiesNote from './notesections/AllergiesNote';
import SocialHistoryNote from './notesections/SocialHistoryNote';
import FamilyHistoryNote from './notesections/FamilyHistoryNote';
import ReviewOfSystemsNote from './notesections/ReviewOfSystemsNote';
import PhysicalExamNote from './notesections/PhysicalExamNote';
import PlanNote from './notesections/PlanNote';
import './GenerateNote.css';

import IdentityForm from '../../../../components/tools/IdentityForm';

import './GenerateNote.css';
import 'pages/EditNote/content/hpi/knowledgegraph/src/css/Button.css';
import { GENERATE_NOTE_MOBILE_BP } from 'constants/breakpoints';
import { PatientPronouns } from 'constants/patientInformation';

interface GenerateNoteProps {
    previousFormClick: () => void;
}

interface DispatchProps {
    updatePatientName: (patientName: string) => UpdatePatientNameAction;
    updatePatientPronouns: (
        pronouns: PatientPronouns
    ) => UpdatePatientPronounsAction;
    updatePatientInformation: (
        patientName: string,
        pronouns: PatientPronouns
    ) => UpdatePatientInformationAction;
}

interface StateProps {
    ROSState: ReviewOfSystemsState;
    planState: PlanState;
    familyHistoryState: FamilyHistoryState;
    socialHistoryState: SocialHistoryState;
    allergiesState: AllergiesState;
    medicationsState: MedicationsState;
    surgicalHistoryState: SurgicalHistoryState;
    medicalHistoryState: MedicalHistoryState;
    physicalExamState: PhysicalExamState;
    patientInformationState: PatientInformationState;
}

const mapStateToProps = (state: CurrentNoteState): StateProps => ({
    ROSState: selectReviewOfSystemsState(state),
    planState: selectPlanState(state),
    familyHistoryState: selectFamilyHistoryState(state),
    socialHistoryState: state.socialHistory,
    allergiesState: selectAllergiesState(state),
    medicationsState: selectMedicationsState(state),
    surgicalHistoryState: selectSurgicalHistoryState(state),
    medicalHistoryState: state.medicalHistory,
    physicalExamState: state.physicalExam,
    patientInformationState: selectPatientInformationState(state),
});

type Props = GenerateNoteProps & StateProps & DispatchProps;

// TODO: look into <li> keys -- throws a warning if duplicats, not a huge deal but probably fix
const GenerateNote: React.FunctionComponent<Props> = (props: Props) => {
    const {
        previousFormClick,
        updatePatientName,
        updatePatientPronouns,
        updatePatientInformation,
        ROSState,
        planState,
        familyHistoryState,
        socialHistoryState,
        allergiesState,
        medicationsState,
        surgicalHistoryState,
        medicalHistoryState,
        physicalExamState,
        patientInformationState,
    } = props;

    const { patientName, pronouns } = patientInformationState;

    // Opening and closing modal with patient information
    const [isModalOpen, setIsModalOpen] = useState(false);
    const openModal = (): void => setIsModalOpen(true);
    const closeModal = (): void => setIsModalOpen(false);
    const savePatientInfo = () => {
        updatePatientInformation(patientName, pronouns);
        closeModal();
    };

    const [isRichText, setIsRichText] = useState(false);

    /** TODO: This logic is loosely duplicated from NavMenu.tsx. This function has been brute-force copied
     * throughout the codebase 10+ times to determine whether or not each individual component is "mobile".
     * Many times the logic uses a different px value to determine what "mobile" is. The "mobile" prop is also
     * prop-drilled through many levels when used. Recomendations to clean this up:
     * 1) There should only be one value for "mobile" in terms of px width across the app.
     * 2) The logic for determining if the app is displaying at the mobile breakpoint should only appear once.
     * 3) Prop-drilling should be avoided. This is a good article that details
     * how to use composition to avoid prop-drilling: https://javascript.plainenglish.io/composition-in-react-f02afe24bc46
     */
    const [isMobile, setIsMobile] = useState(false);
    useEffect(() => {
        const updateDimensions = (): void => {
            const windowWidth =
                typeof window !== 'undefined' ? window.innerWidth : 0;
            setIsMobile(windowWidth < GENERATE_NOTE_MOBILE_BP);
        };
        updateDimensions();
        window.addEventListener('resize', updateDimensions);
        return (): void =>
            window.removeEventListener('resize', updateDimensions);
    }, []);

    const copyNote = () => {
        const note = document.querySelectorAll('.generate-note-text');
        const blob = new Blob([note[0].innerHTML], { type: 'text/html' });
        const clipboardItem = new ClipboardItem({
            ['text/html']: blob,
        });
        navigator.clipboard.write([clipboardItem]);
    };

    const richOrPlainButtons = (
        <Button.Group>
            <Button
                onClick={() => setIsRichText(false)}
                className={`hpi-ph-button${isRichText ? '' : '-selected'}`}
            >
                Plain Text{' '}
            </Button>
            <Button.Or />
            <Button
                onClick={() => setIsRichText(true)}
                className={`hpi-ph-button${isRichText ? '-selected' : ''}`}
            >
                Rich Text
            </Button>
        </Button.Group>
    );

    let generateNoteButtons;
    if (!isMobile) {
        generateNoteButtons = (
            <Fragment>
                <div className={'ui input large'}>
                    <input
                        id='patientName'
                        type='text'
                        autoComplete='off'
                        placeholder="Patient's name (optional)"
                        value={patientName}
                        onChange={(e) => updatePatientName(e.target.value)}
                    ></input>
                </div>
                <Button.Group>
                    <Button
                        onClick={() =>
                            updatePatientPronouns(PatientPronouns.She)
                        }
                        className={`hpi-ph-button${
                            pronouns === PatientPronouns.She ? '-selected' : ''
                        }`}
                    >
                        She
                    </Button>
                    <Button.Or />
                    <Button
                        onClick={() =>
                            updatePatientPronouns(PatientPronouns.They)
                        }
                        className={`hpi-ph-button${
                            pronouns === PatientPronouns.They ? '-selected' : ''
                        }`}
                    >
                        They
                    </Button>
                    <Button.Or />
                    <Button
                        onClick={() =>
                            updatePatientPronouns(PatientPronouns.He)
                        }
                        className={`hpi-ph-button${
                            pronouns === PatientPronouns.He ? '-selected' : ''
                        }`}
                    >
                        He
                    </Button>
                </Button.Group>
                {richOrPlainButtons}
            </Fragment>
        );
    } else {
        generateNoteButtons = (
            <Modal
                className='patient-modal'
                onClose={closeModal}
                onOpen={openModal}
                open={isModalOpen}
                size='tiny'
                dimmer='inverted'
                trigger={
                    <Button basic color='teal' name='home' icon='info circle' />
                }
            >
                <Header>Patient Information</Header>
                <Modal.Content>
                    <Form>
                        <Form.Field
                            fluid
                            autoComplete='off'
                            id='patientName'
                            type='text'
                            label="Patient's Name"
                            className='patient-info-input'
                            placeholder="Enter the patient's name, e.g. Ms. Lee (optional)"
                            control={Input}
                            value={patientName}
                            onChange={(e: any) =>
                                updatePatientName(e.target.value)
                            }
                        />
                        <IdentityForm />
                        {richOrPlainButtons}
                    </Form>
                </Modal.Content>
                <Modal.Actions>
                    <Button
                        color='blue'
                        type='submit'
                        onClick={savePatientInfo}
                        content='Save'
                    />
                </Modal.Actions>
            </Modal>
        );
    }

    return (
        <div>
            <div className='generate-note-buttons'>
                <Segment
                    id='generateNoteButtonsContainer'
                    className='generate-note-buttons__container'
                >
                    {generateNoteButtons}
                    <Button.Group floated='right'>
                        <Button
                            className='copy-button'
                            onClick={() => copyNote()}
                        >
                            Copy Note
                        </Button>
                    </Button.Group>
                </Segment>
            </div>
            <Segment className='generate-note-text'>
                {/* <h1> {this.context.title} </h1> */}
                <h3> History of Present Illness </h3>
                <HPINote />
                <h3> Patient History </h3>
                <h4> Medical History </h4>
                <MedicalHistoryNote
                    isRich={isRichText}
                    medicalHistory={medicalHistoryState}
                />
                <h4> Surgical History </h4>
                <SurgicalHistoryNote
                    isRich={isRichText}
                    surgicalHistory={surgicalHistoryState}
                />
                <h4> Medications </h4>
                <MedicationsNote
                    isRich={isRichText}
                    medications={medicationsState}
                />
                <h4> Allergies </h4>
                <AllergiesNote isRich={isRichText} allergies={allergiesState} />
                <h4> Social History </h4>
                <SocialHistoryNote
                    isRich={isRichText}
                    socialHistory={socialHistoryState}
                />
                <h4> Family History </h4>
                <FamilyHistoryNote
                    isRich={isRichText}
                    familyHistory={familyHistoryState}
                />
                <h3> Review of Systems </h3>
                <ReviewOfSystemsNote isRich={isRichText} ROSState={ROSState} />
                <h3> Physical Exam </h3>
                <PhysicalExamNote
                    isRich={isRichText}
                    physicalExam={physicalExamState}
                />
                <h3> Plan </h3>
                <PlanNote planState={planState} />
            </Segment>
            <Button
                icon
                labelPosition='left'
                floated='left'
                onClick={previousFormClick}
                className='note-previous-button'
            >
                Previous
                <Icon name='arrow left' />
            </Button>
            {/* mobile */}
            <Button
                icon
                floated='left'
                onClick={previousFormClick}
                className='small-note-previous-button'
            >
                <Icon name='arrow left' />
            </Button>
        </div>
    );
};

export default connect(mapStateToProps, {
    updatePatientName,
    updatePatientPronouns,
    updatePatientInformation,
})(GenerateNote);
