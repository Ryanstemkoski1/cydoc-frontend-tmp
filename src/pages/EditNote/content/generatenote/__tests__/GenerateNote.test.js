import React from 'react';
import { shallow } from 'enzyme';
import GenerateNote from '../GenerateNote';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { initialPhysicalExamState } from 'redux/reducers/physicalExamReducer';
import { initialReviewOfSystemsState } from 'constants/reviewOfSystemsInitial';

const mockStore = configureMockStore([thunk]);

const initialStore = {
    reviewOfSystems: initialReviewOfSystemsState,
    physicalExam: initialPhysicalExamState,
    medicalHistory: {},
    surgicalHistory: {},
    medications: {},
    allergies: {},
    socialHistory: {},
    familyHistory: {},
    plan: {},
};

const store = mockStore(initialStore);

describe('Generate Note', () => {
    it('matches snapshot', () => {
        const component = shallow(<GenerateNote store={store} />);
        expect(component).toMatchSnapshot();
    });
});
