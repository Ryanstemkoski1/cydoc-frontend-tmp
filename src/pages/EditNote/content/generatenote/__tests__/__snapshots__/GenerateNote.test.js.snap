// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Generate Note matches snapshot 1`] = `
<ContextProvider
  value={null}
>
  <GenerateNote
    ROSState={
      Object {
        "Cardiovascular/Hematological": Object {
          "Chest pain": "",
          "Easy bleeding": "",
          "Easy bruising": "",
          "Palpitations": "",
        },
        "Dermatologic": Object {
          "Skin changes": "",
        },
        "Eyes/Ears": Object {
          "Hearing changes": "",
          "Vision changes": "",
        },
        "Gastrointestinal": Object {
          "Abdominal pain": "",
          "Constipation": "",
          "Diarrhea": "",
          "Nausea": "",
          "Vomiting": "",
        },
        "General": Object {
          "Chills": "",
          "Fatigue": "",
          "Fever": "",
          "Weight changes": "",
        },
        "Musculoskeletal": Object {
          "Joint pain": "",
          "Muscle pain": "",
        },
        "Neurological": Object {
          "Dizziness": "",
          "Headache": "",
          "Loss of consciousness": "",
          "Numbness": "",
          "Tingling": "",
        },
        "Psychiatric": Object {
          "Mood changes": "",
        },
        "Respiratory": Object {
          "Cough": "",
          "Shortness of breath": "",
        },
        "Urinary": Object {
          "Dysuria": "",
          "Hematuria": "",
          "Urinary frequency": "",
        },
      }
    }
    allergiesState={Object {}}
    familyHistoryState={Object {}}
    medicalHistoryState={Object {}}
    medicationsState={Object {}}
    physicalExamState={
      Object {
        "sections": Object {
          "Cardiac": Object {
            "comments": "",
            "findings": Object {
              "PMI enlarged": false,
              "PMI lateral": false,
              "RRR": false,
              "S3 (ventricular gallop)": false,
              "S4 (presystolic gallop)": false,
              "atrial fibrillation": false,
              "bradycardia": false,
              "carotid bruit": Object {
                "center": false,
                "left": false,
                "right": false,
              },
              "decreased JVP": false,
              "double impulse": false,
              "elevated JVP": false,
              "no bruits": false,
              "no gallops": false,
              "no murmurs": false,
              "no rubs": false,
              "normal JVP": false,
              "normal PMI": false,
              "normal S1 S2": false,
              "pericardial rub": false,
              "sinus arrhythmia": false,
              "sustained PMI": false,
              "tachycardia": false,
            },
          },
          "Ears": Object {
            "comments": "",
            "findings": Object {
              "Rinne unable to still hear fork on left": false,
              "Rinne unable to still hear fork on right": false,
              "Weber louder on left": false,
              "Weber louder on right": false,
              "absent light reflex": Object {
                "center": false,
                "left": false,
                "right": false,
              },
              "diminished auditory acuity": Object {
                "center": false,
                "left": false,
                "right": false,
              },
              "dull light reflex": Object {
                "center": false,
                "left": false,
                "right": false,
              },
              "earwax": Object {
                "center": false,
                "left": false,
                "right": false,
              },
              "exudate": Object {
                "center": false,
                "left": false,
                "right": false,
              },
              "normal Rinne": Object {
                "center": false,
                "left": false,
                "right": false,
              },
              "normal Weber": Object {
                "center": false,
                "left": false,
                "right": false,
              },
              "normal auditory acuity": Object {
                "center": false,
                "left": false,
                "right": false,
              },
              "normal tympanic membrane": Object {
                "center": false,
                "left": false,
                "right": false,
              },
              "otitis externa": Object {
                "center": false,
                "left": false,
                "right": false,
              },
              "otitis media": Object {
                "center": false,
                "left": false,
                "right": false,
              },
              "otoscopy unremarkable": Object {
                "center": false,
                "left": false,
                "right": false,
              },
              "pain on palpation": Object {
                "center": false,
                "left": false,
                "right": false,
              },
              "tenderness to palpation": Object {
                "center": false,
                "left": false,
                "right": false,
              },
              "tympanic membrane bulging": Object {
                "center": false,
                "left": false,
                "right": false,
              },
              "tympanic membrane red": Object {
                "center": false,
                "left": false,
                "right": false,
              },
            },
          },
          "Extremities": Object {
            "comments": "",
            "findings": Object {
              "clubbing": false,
              "cyanosis": false,
              "no clubbing": false,
              "no cyanosis": false,
              "no edema": false,
              "no nail changes": false,
              "onychomycosis": false,
              "pedal edema": false,
            },
          },
          "Eyes": Object {
            "comments": "",
            "findings": Object {
              "EOMI": Object {
                "center": false,
                "left": false,
                "right": false,
              },
              "PERRLA": false,
              "ambylopia": Object {
                "center": false,
                "left": false,
                "right": false,
              },
              "diplopia": Object {
                "center": false,
                "left": false,
                "right": false,
              },
              "fundoscopy unremarkable": Object {
                "center": false,
                "left": false,
                "right": false,
              },
              "hyperopia": Object {
                "center": false,
                "left": false,
                "right": false,
              },
              "injected/red conjunctiva": Object {
                "center": false,
                "left": false,
                "right": false,
              },
              "myopia": Object {
                "center": false,
                "left": false,
                "right": false,
              },
              "no discharge": Object {
                "center": false,
                "left": false,
                "right": false,
              },
              "no redness": Object {
                "center": false,
                "left": false,
                "right": false,
              },
              "pale conjunctiva": Object {
                "center": false,
                "left": false,
                "right": false,
              },
              "presbyopia": Object {
                "center": false,
                "left": false,
                "right": false,
              },
              "purulent discharge": Object {
                "center": false,
                "left": false,
                "right": false,
              },
              "sclera anicteric": Object {
                "center": false,
                "left": false,
                "right": false,
              },
              "scleral icterus": Object {
                "center": false,
                "left": false,
                "right": false,
              },
              "visual acuity intact": Object {
                "center": false,
                "left": false,
                "right": false,
              },
              "visual fields intact": Object {
                "center": false,
                "left": false,
                "right": false,
              },
              "watery discharge": Object {
                "center": false,
                "left": false,
                "right": false,
              },
            },
          },
          "Gastrointestinal": Object {
            "comments": "",
            "findings": Object {
              "abdominal bruit": false,
              "ascites": false,
              "distention": false,
              "fluid wave": false,
              "hepatomegaly": false,
              "hyperactive bowel sounds": false,
              "hypoactive bowel sounds": false,
              "no bruits": false,
              "no guarding": false,
              "no hepatomegaly": false,
              "no masses": false,
              "no rebound tenderness": false,
              "no splenomegaly": false,
              "nondistended": false,
              "nontender": false,
              "normal bowel sounds": false,
              "obesity": false,
              "soft": false,
              "splenomegaly": false,
              "surgical scars": false,
            },
          },
          "General": Object {
            "comments": "",
            "findings": Object {
              "diaphoretic": false,
              "in distress": false,
              "laying still": false,
              "no acute distress": false,
              "pale": false,
              "sitting upright": false,
              "well-appearing": false,
              "well-nourished": false,
              "writhing": false,
            },
          },
          "Head": Object {
            "comments": "",
            "findings": Object {
              "atraumatic": false,
              "normocephalic": false,
              "nuchal rigidity": false,
            },
          },
          "Kidney": Object {
            "comments": "",
            "findings": Object {
              "CVAT present": Object {
                "center": false,
                "left": false,
                "right": false,
              },
              "no CVAT": false,
            },
          },
          "Neck": Object {
            "comments": "",
            "findings": Object {
              "no lymphadenopathy": false,
              "no thryomegaly": false,
              "supple": false,
              "trachea midline": false,
            },
          },
          "Neuro": Object {
            "comments": "",
            "findings": Object {
              "no pronator drift": false,
              "normal CN-V/trigeminal": false,
              "normal CN-VII/facial": false,
              "normal CN-XI/accessory": false,
              "normal Romberg": false,
              "normal UMN": false,
              "normal cerebellum": false,
              "normal eyebrow raise": false,
              "normal facial sensation": false,
              "normal finger-nose": false,
              "normal head turn against resistance": false,
              "normal heel-shin": false,
              "normal jaw opening against resistance": false,
              "normal proprioception": false,
              "normal sensation to light touch": false,
              "normal shoulder shrug against resistance": false,
              "normal smile": false,
              "normal tandem gait": false,
              "normal temperature sensation": false,
              "normal temporalis/masseter clench": false,
              "normal vibration sensation": false,
            },
          },
          "Nose": Object {
            "comments": "",
            "findings": Object {
              "bleeding": false,
              "no sinus tenderness": false,
              "normal internal nose on otoscopy": false,
              "rhinorrhea": false,
              "sinus tenderness to palpation": false,
            },
          },
          "Pulmonary": Object {
            "comments": "",
            "findings": Object {
              "CTAB": false,
              "barrel chest": false,
              "blue lips": false,
              "coughing": false,
              "gurgling": false,
              "kyphosis": false,
              "labored breathing": false,
              "no wheezes rales or rhonchi": false,
              "normal percussion": false,
              "pectus excavatum": false,
              "pursed lips": false,
              "scoliosis": false,
              "tripod position": false,
              "unable to speak in complete sentences": false,
              "use of accessory muscles of breathing": false,
            },
          },
          "Pulses": Object {
            "comments": "",
            "findings": Object {
              "normal brachial": false,
              "normal dorsalis pedis": false,
              "normal radial": false,
              "normal ulnar": false,
            },
          },
          "Skin": Object {
            "comments": "",
            "findings": Object {
              "decreased turgor": false,
              "no tenting": false,
              "no visible lesions": false,
              "normal turgor": false,
              "tenting": false,
              "warm and dry": false,
            },
          },
          "Tendon Reflexes": Object {
            "comments": "",
            "findings": Object {
              "normal ankle jerk": false,
              "normal biceps": false,
              "normal brachioradialis": false,
              "normal patellar": false,
              "normal plantar": false,
              "normal triceps": false,
            },
          },
          "Throat": Object {
            "comments": "",
            "findings": Object {
              "MMM": false,
              "mucous membranes dry": false,
              "normal phonation": false,
              "normal swallowing": false,
              "oropharynx clear": false,
              "oropharynx erethematous": false,
              "palate lower on left": false,
              "palate lower on right": false,
              "symmetric palate elevation": false,
              "tongue deviates left": false,
              "tongue deviates right": false,
              "tongue pink and moist": false,
              "tongue protrudes midline": false,
            },
          },
        },
        "vitals": Object {
          "RR": 0,
          "diastolicBloodPressure": 0,
          "heartRate": 0,
          "oxygenSaturation": 0,
          "systolicBloodPressure": 0,
          "temperature": 0,
        },
        "widgets": Object {
          "abdomen": Object {
            "leftLowerQuadrant": Object {
              "guarding": false,
              "rebound": false,
              "tenderness": false,
            },
            "leftUpperQuadrant": Object {
              "guarding": false,
              "rebound": false,
              "tenderness": false,
            },
            "rightLowerQuadrant": Object {
              "guarding": false,
              "rebound": false,
              "tenderness": false,
            },
            "rightUpperQuadrant": Object {
              "guarding": false,
              "rebound": false,
              "tenderness": false,
            },
          },
          "lungs": Object {
            "leftLowerLobe": Object {
              "bronchialBreathSounds": false,
              "egophony": false,
              "rales": false,
              "rhonchi": false,
              "stridor": false,
              "vesicularBreathSounds": false,
              "wheezes": false,
              "whistling": false,
            },
            "leftUpperLobe": Object {
              "bronchialBreathSounds": false,
              "egophony": false,
              "rales": false,
              "rhonchi": false,
              "stridor": false,
              "vesicularBreathSounds": false,
              "wheezes": false,
              "whistling": false,
            },
            "lingula": Object {
              "bronchialBreathSounds": false,
              "egophony": false,
              "rales": false,
              "rhonchi": false,
              "stridor": false,
              "vesicularBreathSounds": false,
              "wheezes": false,
              "whistling": false,
            },
            "rightLowerLobe": Object {
              "bronchialBreathSounds": false,
              "egophony": false,
              "rales": false,
              "rhonchi": false,
              "stridor": false,
              "vesicularBreathSounds": false,
              "wheezes": false,
              "whistling": false,
            },
            "rightMiddleLobe": Object {
              "bronchialBreathSounds": false,
              "egophony": false,
              "rales": false,
              "rhonchi": false,
              "stridor": false,
              "vesicularBreathSounds": false,
              "wheezes": false,
              "whistling": false,
            },
            "rightUpperLobe": Object {
              "bronchialBreathSounds": false,
              "egophony": false,
              "rales": false,
              "rhonchi": false,
              "stridor": false,
              "vesicularBreathSounds": false,
              "wheezes": false,
              "whistling": false,
            },
          },
          "murmurs": Object {},
          "pulses": Object {},
          "reflexes": Object {},
        },
      }
    }
    socialHistoryState={Object {}}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    surgicalHistoryState={Object {}}
    updatePatientInformation={[Function]}
    updatePatientName={[Function]}
    updatePatientPronouns={[Function]}
  />
</ContextProvider>
`;
